core/src/main/java/io/questdb/griffin/engine/functions/regex/AbstractLikeSymbolFunctionFactory.java

```diff
@@ -184,31 +184,47 @@ public Function getRight() {
         @Override
         public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) throws SqlException {
             BinaryFunction.super.init(symbolTableSource, executionContext);
+            if (stateInherited) {
+                return;
+            }
+            this.stateShared = false;
             // this is bind variable, we can use it as constant
             final CharSequence patternValue = pattern.getStrA(null);
             if (patternValue != null && patternValue.length() > 0) {
+                // lastPattern is used to avoid recompiling the same regex multiple times on
+                // different cursor invocations
                 String p = escapeSpecialChars(patternValue, lastPattern);
                 if (p != null) {
                     int flags = Pattern.DOTALL;
                     if (caseInsensitive) {
                         flags |= Pattern.CASE_INSENSITIVE;
                         p = p.toLowerCase();
                     }
-                    this.matcher = Pattern.compile(p, flags).matcher("");
+                    Matcher matcher = Pattern.compile(p, flags).matcher("");
                     this.lastPattern = p;
+                    extractSymbolKeys(value, symbolKeys, matcher);
                 }
             } else {
                 lastPattern = null;
-                matcher = null;
             }
-            initialized = false;
         }
 
         @Override
         public boolean isThreadSafe() {
             return false;
```

Comment:
nit: we could change this to
```suggestion
            return value.isThreadSafe();
```
or even `return true;` (the latter is since we call `rec.getInt()` on a column with static symbol table).

The same suggestion applies to other static symbol table functions here.

Reply:
can't do "true", getInt()  could be a string parsing thing or some other contraption

Reply:
fixed here: https://github.com/questdb/questdb/pull/5445/commits/d9524e8749e659c8740ed5a22dfadbb38e5d6b0e

Reply:
> can't do "true", getInt() could be a string parsing thing or some other contraption

In that case it wouldn't have a static symbol table. Only symbol columns have it.